crates/solver/src/metrics.rs

```diff
@@ -41,10 +41,14 @@ pub enum SolverRunOutcome {
 pub enum SettlementSubmissionOutcome {
     /// A settlement transaction was mined and included on the blockchain.
     Success,
+    /// A settlement transaction was mined and included on the blockchain with a failed status.
+    MinedFailed,
```

Comment:
very nit-picky naming-nit: To me, its more natural for `Revert` to indicated that a transaction got mined and reverted. Originally, I had intended for `Failure` to be used for this:

> A transaction failed to be submitted or, in the case of private network submission, the blockchain state changed and the transaction is no longer valid.

That being said maybe we should differentiate between "failed to submit" and "transaction started simulating failure" and have a `SimulationRevert` just for simulations?

Reply:
With that in mind, we should probably update the documentation for the `Failure` variant as it doesn't seem accurate to me.